web/src/routes/OAuth.tsx

```diff
@@ -0,0 +1,16 @@
+import React, { ReactElement, useEffect } from 'react'
+import { Redirect, useLocation } from 'react-router-dom'
+import { useLocalStorageState } from '@umijs/hooks'
+
+export default function OAuth(): ReactElement {
+  const location = useLocation()
+  const params = new URLSearchParams(location.search)
```

Comment:
Might be better to move `const params = new URLSearchParams(location.search)` inside of useEffect and set a dependency for `location.search`. This way it won't create `new URLSearchParams(location.search)` on each render